File: YuGiOh2/Cards/C01248895.cs
using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 连锁破坏
    /// </summary>
    public class C01248895
    {
        public static int Type { get; set; } = (int)AffectMomentType.WhenSummoned;

        public static void ProcessEffect(string targetID, Player player)
        {
            for (int i = 0; i < 5; i++)
            {
                if (player.Enemy.Field.MonsterFields[i] == null ||
                    player.Enemy.Field.MonsterFields[i].UID != targetID ||
                    player.Enemy.Field.MonsterFields[i].ATK > 2000)
                    continue;

                var result = player.Enemy.Deck.Where(c => c.Password == player.Enemy.Field.MonsterFields[i].Password).ToList();
                for (int j = result.Count - 1; j >= 0 ; j--)
                {
                    var card = result[j];
                    player.Enemy.AddCardToGrave(ref card);
                    player.Enemy.Deck.RemoveAt(j);
                }
                player.Enemy.AddCardToGrave(ref player.Enemy.Field.MonsterFields[i]);
                player.Enemy.Field.MonsterFields[i] = null;
            }
        }
    }
}

File: YuGiOh2/Scripts/C01248895.lua
Type = 1;
--连锁破坏
function processEffect(targetID, player)
    for index, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and monster.UID == targetID and monster.ATK <= 2000 then
            local pwd = monster.Password;
            local tmpDeck = {};
            player.Enemy:AddCardToGrave(monster);
            player.Enemy:ClearMonsterField(index - 1);
            for i, v in ipairs(player.Enemy.Deck) do
                if v.Password ~= pwd then
                    table.insert(tmpDeck, v);
                else
                    player.Enemy:AddCardToGrave(v);
                end
            end
            player.Enemy.Deck = tmpDeck;
        else
            table.insert(tmpMonster, monster);
        end
    end
end